qiskit/transpiler/passes/layout/noise_adaptive_layout.py

```diff
@@ -73,6 +73,7 @@ def __init__(self, backend_prop):
         self.gate_reliability = {}
         self.swap_reliabs = {}
         self.prog_graph = rx.PyGraph()
+        self.prog_neighbors = {}
```

Comment:
Should this be reset (during `run` or similar) so that subsequent calls to `run` don't start with invalid state? (Similar to the issue discussed here: https://github.com/Qiskit/qiskit-terra/issues/4984 )

Reply:
yep, good call. I'll move that there. I just put it here because all the initialization for things were here

Reply:
Should I move all the initializations there?

Reply:
Yes, unless there are any that would make sense to keep in state across runs (like https://github.com/Qiskit/qiskit-terra/blob/1a3cd4c/qiskit/transpiler/passes/optimization/commutation_analysis.py#L37 ), but these are rare.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5146/commits/ab1d3a738b276b34030a44b2a1d22319e01f1bd0